File: day2/python/solution.py
with open('../input.txt', 'r') as handle:
    text = handle.read().splitlines()
    i = 0
    length = len(text)
    depth = 0
    horizontal = 0
    while i < length:
        [type, number] = text[i].split(' ')
        number = int(number)
        if type == 'forward':
            horizontal += number
        elif type == 'down':
            depth += number
        elif type == 'up':
            depth -= number
        i += 1
        
    print(horizontal * depth)

File: day2/ruby/solution.rb
file = File.open('input.txt')
file_data = file.readlines.map(&:chomp)
horizontal = 0
depth = 0
(0..file_data.length).each do |i|
  value, number = file_data[i].to_s.split(/ /)
  case value
  when "forward"
    horizontal += number.to_i
  when "down"
    depth += number.to_i
  when "up"
    depth -= number.to_i
  end
end
puts horizontal * depth
